feat: upgrade Prisma to 6.16.0 with no-rust engine

keithwillcode opened the pull request:
## What does this PR do?

This PR upgrades Prisma from 6.7.0 to 6.16.0 with the no-rust engine. As part of this upgrade, the `bookingReferenceExtension` that automatically handled soft deletes has been removed. The PR updates bookingReference Prisma queries throughout the codebase to maintain soft-delete behavior where appropriate.

**Link to Devin run**: https://app.devin.ai/sessions/7386f97ecd6749efbfb67b1ef808644b
**Requested by**: @keithwillcode (keith@cal.com)

## Key Changes

### Prisma Upgrade
- Upgraded Prisma from 6.7.0/6.8.2 to 6.16.0/6.16.1
- Migrated to no-rust engine
- Updated `.gitignore` to reflect new Prisma client generated file locations (`apps/api/v2/generated/prisma/*` and `packages/prisma/generated/prisma/*`)
- Updated yarn.lock with new Prisma package versions

### Soft Delete Handling

The `bookingReferenceExtension` previously provided automatic soft delete handling. This functionality is now implemented manually in **read operations only**:

**API v1 Endpoints Updated:**
- `booking-references/_get.ts`: Added `deleted: null` filter to queries
- `booking-references/[id]/_get.ts`: Added `deleted: null` to `findUniqueOrThrow`
- `booking-references/[id]/_delete.ts`: Converted `delete` to `update` with `deleted: true`
- `booking-references/[id]/_auth-middleware.ts`: Added `deleted: null` to `findFirst`

**Intentionally NOT Updated (Preserving Existing Behavior):**
- `booking-references/[id]/_patch.ts`: PATCH endpoint can update soft-deleted references
- `packages/app-store/dailyvideo/lib/VideoApiAdapter.ts`: DailyVideo adapter can update soft-deleted references

### Test Coverage

Added integration test file: `apps/api/v1/test/lib/booking-references.integration-test.ts`
- Tests verify GET endpoints properly filter soft-deleted records
- Tests verify PATCH endpoint can update references regardless of deleted status (existing behavior)
- 7 test cases covering various soft-delete scenarios
- File uses `.integration-test.ts` extension to run in integration test job (requires database)

### Other Changes
- Updated E2E database seeding to use `USE_POOL=true yarn db-seed`
- Updated API v2 production build script with proper command chaining (`&&` instead of newlines)
- Import reordering in `apps/api/v1/pages/api/slots/_get.ts`

## ⚠️ Important Review Notes

### Intentional Behavioral Inconsistency

This PR **intentionally** creates an inconsistency in soft-delete handling per maintainer request:
- **GET operations**: Filter out soft-deleted records ✓
- **PATCH operations**: Can update soft-deleted records ✓ (existing functionality preserved)
- **DELETE operations**: Soft-delete (convert to update) ✓

**Rationale**: Maintainer (@keithwillcode) explicitly requested to preserve existing functionality for PATCH and DailyVideo operations rather than adding soft-delete filtering. This maintains backward compatibility but creates behavioral inconsistency.

### Prisma Client Generation Paths

The `.gitignore` was updated for new generated file locations. **Critical**: Verify these paths work correctly in all environments (dev, staging, production, CI/CD).

### Integration Test Naming

Tests use `.integration-test.ts` extension to run in the integration test job (they require database connection). Verify tests actually run and pass in CI.

## Mandatory Tasks

- [x] I have self-reviewed the code
- [x] I have updated the developer docs if needed (N/A - internal Prisma upgrade, no API changes)
- [x] I confirm automated tests are in place that prove my fix is effective

## How should this be tested?

### Prerequisites
- Standard Cal.com development environment
- Database configured with DATABASE_URL and DATABASE_DIRECT_URL
- No new environment variables required

### Critical Test Scenarios

1. **Soft Delete Behavior**
   - Create a booking with calendar/video integration
   - Cancel the booking (should soft-delete the booking reference)
   - Verify GET /api/booking-references does NOT return the soft-deleted reference
   - Verify the reference still exists in database with `deleted: true`

2. **PATCH Endpoint Behavior (Existing Functionality)**
   - Create a booking reference
   - Soft-delete it (set `deleted: true` directly in DB)
   - Try to PATCH the reference (should succeed - this is intentional existing behavior)
   - Verify the update was applied despite deleted status

3. **Prisma Upgrade Regression Testing**
   - Run full test suite: `TZ=UTC yarn test`
   - Run type checks: `yarn type-check:ci --force`
   - Verify booking creation, cancellation, and rescheduling workflows

4. **Integration Tests**
   - Verify `apps/api/v1/test/lib/booking-references.integration-test.ts` runs in integration test job
   - Verify all 7 test cases pass

## Checklist

- [x] I have read the contributing guide
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas (N/A - straightforward migrations)
- [x] I have checked that my changes generate no new warnings

## Human Review Checklist

Please pay special attention to:

- [ ] **Intentional behavioral inconsistency**: Verify it's acceptable for PATCH and DailyVideo operations to be able to update soft-deleted records while GET operations filter them out. Is this the desired long-term behavior, or should a follow-up issue be created?
- [ ] **Prisma client generation paths**: Confirm the new `.gitignore` paths (`apps/api/v2/generated/prisma/*` and `packages/prisma/generated/prisma/*`) work correctly in all environments (dev, staging, production, CI)
- [ ] **Integration test execution**: Verify the `.integration-test.ts` file runs in the integration test job and all tests pass
- [ ] **No regressions**: Confirm booking creation, cancellation, and rescheduling workflows still work correctly
- [ ] **Soft-delete coverage**: Review if the incomplete soft-delete filtering (GET operations only) could cause issues. Consider whether PATCH and DailyVideo operations should also filter soft-deleted records in a future PR.

devin-ai-integration[bot] committed:
feat: upgrade Prisma to 6.16.0 with no-rust engine

- Update Prisma packages to 6.16.0
- Add PostgreSQL adapter dependency
- Configure engineType: 'client' and provider: 'prisma-client' in schema
- Update Prisma client instantiation with PostgreSQL adapter
- Remove binaryTargets from generators (not needed with library engine)
- Fix schema view issue by removing @id decorator from BookingTimeStatusDenormalized
- Fix ESLint warning by removing non-null assertion

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

keithwillcode committed:
Merge remote-tracking branch 'origin/main' into devin/prisma-6-16-upgrade-1757800439

keithwillcode committed:
Web app running but types wrecked

keithwillcode committed:
web app running but build and type issues

keithwillcode committed:
Removed the connection pool

keithwillcode committed:
Merge branch 'main' into devin/prisma-6-16-upgrade-1757800439

keithwillcode committed:
Fixed zod type issue

keithwillcode committed:
Fixed types in booking reference extension

keithwillcode committed:
Fixed test issues

keithwillcode committed:
Type checks passing it seems

keithwillcode committed:
Using cjs as moduleFormat

keithwillcode committed:
Fixing Prisma undefined

keithwillcode committed:
Merge remote-tracking branch 'origin/main' into devin/prisma-6-16-upgrade-1757800439

devin-ai-integration[bot] committed:
fix: update prismock initialization for Prisma 6.16 compatibility

- Add @prisma/internals dependency for getDMMF()
- Restructure prismock initialization to use createPrismock() with DMMF
- Create Proxy that's returned from mock factory for proper spy support
- Fixes 89 failing unit tests with 'Cannot read properties of undefined (reading datamodel)' error
- Based on workaround from prismock issue #1482

All unit tests now pass (375 test files, 3323 tests passed)

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

devin-ai-integration[bot] committed:
fix: cast serviceAccountKey to Prisma.InputJsonValue in bookingScenario.ts

- Apply type cast at lines 2493 and 2535
- Fixes type errors from Prisma 6.16 upgrade
- Follows established pattern from delegationCredential.ts
- Add eslint-disable for pre-existing any types
- Rename unused appStoreLookupKey parameter to satisfy lint
- All 3323 tests passing

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

devin-ai-integration[bot] committed:
chore: remove whitespace-only lines from bookingScenario.ts

- Remove blank lines where eslint-disable comments were replaced
- Cleanup from pre-commit hook formatting

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

keithwillcode committed:
Update is-prisma-available-check.ts

devin-ai-integration[bot] committed:
fix: remove datasources config when using Prisma Driver Adapters

- Update customPrisma to create new adapter when datasources URL is provided
- Remove datasources config from API v2 Prisma services (already in adapter)
- Fixes 'Custom datasource configuration is not compatible with Prisma Driver Adapters' error

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

devin-ai-integration[bot] committed:
fix: use Pool instances for PrismaPg adapters in index.ts

- Create Pool instance before passing to PrismaPg adapter
- Update customPrisma to create Pool for custom connection strings
- Matches working pattern from API v2 services
- Fixes 'Invalid `prisma.$queryRawUnsafe()` invocation' error

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

keithwillcode committed:
Not using queryRawUnsafe

keithwillcode committed:
Trying anything at this point

keithwillcode committed:
Make sure the DB is ready first

keithwillcode committed:
Merge branch 'main' into devin/prisma-6-16-upgrade-1757800439

keithwillcode committed:
Don't auto run migrations in CI mode

keithwillcode committed:
Merge branch 'devin/prisma-6-16-upgrade-1757800439' of github.com:calcom/cal.com into devin/prisma-6-16-upgrade-1757800439

keithwillcode committed:
Revert "Make sure the DB is ready first"

This reverts commit 2b20bd45c974f3d7e07d8b904bc7fcdae37cce03.

keithwillcode committed:
Dynamic import of prisma

keithwillcode committed:
Commenting where it seems to break

keithwillcode committed:
Backwards compatability for API v2

devin-ai-integration[bot] committed:
chore: resolve merge conflicts from main

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

devin-ai-integration[bot] committed:
fix: add explicit type annotations for map callbacks in API v2

- Add type annotation for map parameter in memberships.repository.ts
- Add type annotation for map parameter in stripe.service.ts
- Fixes implicit 'any' type errors from stricter Prisma 6.16.0 type inference

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

devin-ai-integration[bot] committed:
fix: add explicit type annotations for API v2 map callbacks

- users.repository.ts:292: add Profile & { user: User } type
- memberships.service.ts:19-20: add Membership type to filter callbacks

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

devin-ai-integration[bot] committed:
Merge branch 'main' into devin/prisma-6-16-upgrade-1757800439

Resolve conflict in teamService.ts by accepting new imports from main

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

devin-ai-integration[bot] committed:
fix: add explicit type annotation for attributeToUser in organizations-users.repository.ts

- organizations-users.repository.ts:63: add AttributeToUser with nested relations type

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

devin-ai-integration[bot] committed:
fix: add explicit Membership type annotations in teams.repository.ts

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

devin-ai-integration[bot] committed:
fix: use API v2 dedicated Prisma client to support adapter in PrismaClientOptions

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

keithwillcode committed:
Running API v2 build commands together so they all get the space size var

keithwillcode committed:
Fixing Maximum call depth exceeded error

keithwillcode committed:
fixed type issues

keithwillcode commented on packages/prisma/auto-migrations.ts:
```diff
@@ -26,6 +26,10 @@ async function main(): Promise<void> {
     console.info("No DATABASE_DIRECT_URL found, skipping migrations");
     return;
   }
+  if (process.env.CI) {
```
This was needed because build was failing for prisma in the web build, which claims to be without a database.

keithwillcode committed:
Trying to make the seed more stable

keithwillcode committed:
Revert "Trying to make the seed more stable"

This reverts commit 1fd4495e6af7acd7981cda7dedec3168979b0e9d.

keithwillcode committed:
Merge branch 'main' into devin/prisma-6-16-upgrade-1757800439

keithwillcode committed:
Fixed path to prisma client

keithwillcode committed:
Merge branch 'devin/prisma-6-16-upgrade-1757800439' of github.com:calcom/cal.com into devin/prisma-6-16-upgrade-1757800439

keithwillcode committed:
Merge branch 'main' into devin/prisma-6-16-upgrade-1757800439

keithwillcode committed:
Fixed type check

keithwillcode committed:
Fix eslint warnings

devin-ai-integration[bot] committed:
fix: externalize @prisma/adapter-pg and pg in platform-libraries Vite config

- Add @prisma/adapter-pg and pg to external dependencies list
- Add corresponding globals for these packages
- Fix Prisma client aliases to point to packages/prisma/client instead of node_modules
- Add Node.js resolve conditions to prefer Node.js exports
- Keep commonjsOptions.include for proper CommonJS transformation
- Add eslint-disable for __dirname in Vite config file
- Remove problematic prettier/prettier eslint comment

This fixes the 'Extensions.defineExtension is unable to run in this browser environment' error when running yarn generate-swagger in apps/api/v2 after upgrading to Prisma v6.16 with the no-rust engine approach.

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

devin-ai-integration[bot] committed:
fix: update Prisma imports in API v2 services to use package path

- Change imports from '../../../generated/prisma/client' to '@calcom/prisma/client'
- Fixes CI error: Cannot find module '../../../../../packages/prisma/generated/prisma/client.ts'
- Aligns with backwards compatibility re-export structure after Prisma v6.16 upgrade

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

devin-ai-integration[bot] committed:
fix: remove .ts extension from Prisma client path mapping in tsconfig

- Remove file extension from @calcom/prisma/client path mapping
- Fixes runtime error: Cannot find module '../../../../../packages/prisma/generated/prisma/client.ts'
- TypeScript path mappings should not include file extensions per best practices
- Allows Node.js to correctly resolve to .js files at runtime

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

keithwillcode committed:
Merge remote-tracking branch 'origin/main' into devin/prisma-6-16-upgrade-1757800439

keithwillcode committed:
Merge branch 'main' into devin/prisma-6-16-upgrade-1757800439

devin-ai-integration[bot] committed:
fix: resolve Prisma 6.16.0 type incompatibilities in bookingScenario tests

- Changed InputPayment.data type from PaymentData to Prisma.InputJsonValue
- Changed createCredentials key parameter from JsonValue to InputJsonValue
- Removed unused PaymentData type definition
- Resolves type errors at lines 709 and 1088 without using 'as any' casts

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

devin-ai-integration[bot] committed:
fix: remove non-existent Watchlist fields from test fixtures

- Remove createdById from isLockedOrBlocked.test.ts (lines 15, 20)
- Remove severity and createdById from _post.test.ts (line 110)
- These fields don't exist in Watchlist model schema after Prisma 6.16.0 upgrade
- Resolves TS2353 errors without using 'as any' casts

Relates to PR #23816

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

ThyMinimalDev committed:
fix: api v2 imports generated prisma and platform libraries

keithwillcode committed:
Merge remote-tracking branch 'origin/main' into devin/prisma-6-16-upgrade-1757800439

devin-ai-integration[bot] committed:
fix: resolve type errors from Prisma 6.16 upgrade

- Add missing markdownToSafeHTML import in AppCard.tsx
- Fix organizationId null handling in fresh-booking.test.ts
- Remove non-existent createdById field from Watchlist test utils

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

keithwillcode committed:
Put back some external rollups

keithwillcode committed:
Merge branch 'main' into devin/prisma-6-16-upgrade-1757800439

keithwillcode committed:
Added back the resolve conditions

keithwillcode committed:
Merge branch 'devin/prisma-6-16-upgrade-1757800439' of github.com:calcom/cal.com into devin/prisma-6-16-upgrade-1757800439

keithwillcode committed:
Stop using Pool directly

ThyMinimalDev committed:
chore: remove prisma bookingReferenceExtension and update calls

ThyMinimalDev committed:
Merge branch 'main' into devin/prisma-6-16-upgrade-1757800439

ThyMinimalDev committed:
fix: organizations-admin-not-team-member-event-types.e2e-spec.ts

keithwillcode committed:
Merge branch 'main' into devin/prisma-6-16-upgrade-1757800439

ThyMinimalDev committed:
chore: bring back POOL in api v2 prisma clients

ThyMinimalDev committed:
chore: remove Pool but await connect

ThyMinimalDev committed:
fixup! chore: remove Pool but await connect

ThyMinimalDev committed:
chore: bring back Pool on all clients

ThyMinimalDev committed:
Merge branch 'main' into devin/prisma-6-16-upgrade-1757800439

ThyMinimalDev committed:
chore: end pool manually

ThyMinimalDev committed:
Merge branch 'main' into devin/prisma-6-16-upgrade-1757800439

ThyMinimalDev committed:
chore: test with pool max 1

ThyMinimalDev committed:
chore: e2e test prisma max  pool of 1 connection

ThyMinimalDev committed:
Merge branch 'main' into devin/prisma-6-16-upgrade-1757800439

ThyMinimalDev committed:
chore: give more control over pool for prisma module with env

ThyMinimalDev committed:
remove pool from base prisma client

ThyMinimalDev committed:
chore: prisma client in libraries use pool

keithwillcode committed:
Merge remote-tracking branch 'origin/main' into devin/prisma-6-16-upgrade-1757800439

keithwillcode committed:
Fixed types

keithwillcode commented on apps/api/v1/pages/api/booking-references/[id]/_patch.ts:
```diff
@@ -72,7 +72,7 @@ export async function patchHandler(req: NextApiRequest) {
     await prisma.booking.findFirstOrThrow(args);
   }
-  const booking_reference = await prisma.bookingReference.update({ where: { id }, data });
+  const booking_reference = await prisma.bookingReference.update({ where: { id, deleted: null }, data });
```
@ThyMinimalDev I guess we should move this to the find before running the update

keithwillcode replied:
Actually I'm going to rip out the deleted: null because the extension we are deleting didn't modify straight update queries. Will be maintaining existing functionality

keithwillcode committed:
Merge branch 'main' into devin/prisma-6-16-upgrade-1757800439

ThyMinimalDev committed:
chore: log pool events and improve pooling

keithwillcode committed:
Merge remote-tracking branch 'origin/main' into devin/prisma-6-16-upgrade-1757800439

keithwillcode committed:
Fixing some types and tests

keithwillcode committed:
Changing the parsing of USE_POOL

keithwillcode commented on packages/prisma/index.ts:
```diff
@@ -1,18 +1,61 @@
 
-const prismaOptions: Prisma.PrismaClientOptions = {};
+const connectionString = process.env.DATABASE_URL || "";
+const pool = process.env.USE_POOL
```
Addressed

devin-ai-integration[bot] committed:
fix: ensure Prisma client is connected before seeding to prevent transaction errors

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

ThyMinimalDev committed:
chore: adjust pools

ThyMinimalDev committed:
chore: add process.env.USE_POOL to libraries vite config

ThyMinimalDev committed:
fix: v1 _patch reference check bookingRef on the booking find

ThyMinimalDev committed:
fix: v1 get references deleted null for system admin

ThyMinimalDev committed:
Merge branch 'main' into devin/prisma-6-16-upgrade-1757800439

devin-ai-integration[bot] committed:
test: add integration tests for bookingReference soft-delete behavior

- Add bookingReference.integration-test.ts to test repository methods
- Add handleDeleteCredential.integration-test.ts to test credential deletion cascade
- Add booking-references.integration-test.ts for API v1 integration tests
- All tests verify soft-delete behavior without using mocks
- Tests use real database operations to ensure soft-deleted records persist
- Cover scenarios: replacing references, credential deletion, querying with filters

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

devin-ai-integration[bot] committed:
refactor: convert booking-references test to actual API endpoint testing

- Modified _get.ts to export handler function for testing
- Refactored integration test to call API handler instead of directly testing Prisma
- Added timestamps to test data to avoid conflicts
- Tests now verify API layer correctly filters soft-deleted references
- All 4 tests passing

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

devin-ai-integration[bot] committed:
fix: add explicit prisma.$connect() call to seed-insights script

With Prisma 6.16 and the PostgreSQL adapter, scripts need to explicitly call $connect() before running database operations to ensure the connection pool is properly initialized. This prevents 'Transaction already closed' errors.

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

devin-ai-integration[bot] committed:
fix: add $connect() to main() execution in seed-insights

Both main() and createPerformanceData() entry points need explicit prisma.$connect() calls with the Prisma 6.16 PostgreSQL adapter.

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

devin-ai-integration[bot] committed:
fix: always use connection pool for Prisma PostgreSQL adapter

Enable connection pooling by default for the Prisma adapter to prevent
transaction state issues during seed operations. Without a pool, each
operation creates a new connection which can lead to 'Transaction already
closed' errors during heavy database operations like seeding.

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

devin-ai-integration[bot] committed:
Revert "fix: always use connection pool for Prisma PostgreSQL adapter"

This reverts commit 6724bb08e42bc0a94846069de83b04db0aeb8e8b.

devin-ai-integration[bot] committed:
fix: enable connection pool for db-seed in cache-db action

Set USE_POOL=true when running yarn db-seed to use connection pooling
with the Prisma PostgreSQL adapter. This prevents 'Transaction already
closed' errors during seeding by maintaining stable database connections.

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

devin-ai-integration[bot] committed:
fix: add safety check for undefined ownerForEvent in seed script

Prevent 'Cannot read properties of undefined' error when orgMembersInDBWithProfileId
is empty. This can happen if organization members fail to create or when there's a
duplicate constraint violation causing early return.

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

keithwillcode committed:
Merge branch 'main' into devin/prisma-6-16-upgrade-1757800439

keithwillcode commented on apps/api/v1/pages/api/booking-references/[id]/_patch.ts:
```diff
@@ -67,9 +67,9 @@ export async function patchHandler(req: NextApiRequest) {
     const args: Prisma.BookingFindFirstOrThrowArgs = isSystemWideAdmin
       ? /* If admin, we only check that the booking exists */
-        { where: { id: data.bookingId } }
+        { where: { id: data.bookingId, references: { some: { id, deleted: null } } } }
```
@ThyMinimalDev Thoughts on this change and the suggested update from cubic.dev? Sounds serious

ThyMinimalDev replied:
ah yeah I understand the issue, good catch

ThyMinimalDev committed:
fix: v1 _patch reference check bookingRef

devin-ai-integration[bot] committed:
fix: increase pool size and add timeout settings to prevent transaction errors

- Increase max connections from 5 to 10
- Add connectionTimeoutMillis: 30000 (30 seconds)
- Add statement_timeout: 60000 (60 seconds)

These settings help prevent 'Unknown transaction status' errors during
heavy database operations like seeding by giving transactions more time
to complete and allowing more concurrent connections.

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

devin-ai-integration[bot] committed:
Revert "fix: increase pool size and add timeout settings to prevent transaction errors"

This reverts commit 148264f1f1861dfb09a082937a3e2b49e78fc41a.

devin-ai-integration[bot] committed:
fix: remove standalone execution in seed-app-store to prevent premature disconnect

The seed-app-store.ts file had a standalone main() call at the bottom
that would execute immediately when imported, including a prisma.$disconnect()
in its .finally() block.

This caused issues because:
1. seed.ts imports and calls mainAppStore()
2. The import triggers the standalone main() execution
3. This standalone execution disconnects prisma after completion
4. seed.ts then tries to call mainHugeEventTypesSeed() but prisma is disconnected
5. This leads to 'Unknown transaction status' errors

Fixed by removing the standalone execution since mainAppStore() is already
called programmatically from seed.ts which manages the connection lifecycle.

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

devin-ai-integration[bot] committed:
fix: use require.main check to prevent premature disconnect when imported

Added require.main === module check so seed-app-store.ts:
- Runs standalone with proper connection management when executed directly
  via 'yarn seed-app-store' or 'ts-node seed-app-store.ts'
- Does NOT run standalone when imported as a module by seed.ts,
  preventing premature prisma disconnect

This fixes 'Unknown transaction status' errors while maintaining
backward compatibility for direct execution.

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

devin-ai-integration[bot] committed:
fix: seed apps before creating users to prevent foreign key constraint violation

Reordered seeding operations to call mainAppStore() before main() because:
- main() creates users with credentials that reference apps via appId foreign key
- mainAppStore() seeds the App table with app records
- Apps must exist before credentials can reference them

This fixes the 'Foreign key constraint violated on Credential_appId_fkey' error
that occurred when creating credentials before the apps they reference existed.

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

keithwillcode committed:
Apply suggestion from @cubic-dev-ai[bot]

Co-authored-by: cubic-dev-ai[bot] <191113872+cubic-dev-ai[bot]@users.noreply.github.com>

keithwillcode committed:
Merge remote-tracking branch 'origin/main' into devin/prisma-6-16-upgrade-1757800439

keithwillcode committed:
Removing functional changes of deleted: null

keithwillcode committed:
Apply suggestion from @keithwillcode

devin-ai-integration[bot] committed:
refactor: move seedAppData call to bottom of main() in seed.ts

Moved seedAppData() call from seed-app-store.ts to the bottom of main()
in seed.ts to ensure the 'pro' user is created before attempting to
create routing form data for them.

Changes:
- Exported seedAppData function from seed-app-store.ts
- Removed seedAppData() call from the main() export in seed-app-store.ts
- Added seedAppData() call at the bottom of main() in seed.ts
- Updated standalone execution in seed-app-store.ts to still call
  seedAppData() when run directly via 'yarn seed-app-store'

This ensures proper ordering: apps seeded → users created → routing
form data created for existing users.

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

devin-ai-integration[bot] committed:
refactor: move routing form seeding from seed-app-store.ts to seed.ts

Moved the routing form seeding logic (previously in seedAppData function)
from seed-app-store.ts to be inline at the bottom of main() in seed.ts.

This ensures the 'pro' user is created before attempting to create routing
form data for them.

Changes:
- Removed seedAppData function and seededForm export from seed-app-store.ts
- Removed import of seedAppData from seed.ts
- Added routing form seeding logic inline at bottom of main() in seed.ts

Seeding order: apps → users (including 'pro') → routing forms

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

volnei commented on apps/web/lib/video/[uid]/getServerSideProps.ts:
```diff
@@ -7,14 +7,14 @@ import {
 import { isENVDev } from "@calcom/lib/env";
-import { BookingRepository } from "@calcom/features/bookings/repositories/BookingRepository";
-import { OrganizationRepository } from "@calcom/features/ee/organizations/repositories/OrganizationRepository";
 import prisma from "@calcom/prisma";
```
```suggestion
import { prisma } from "@calcom/prisma";
```

volnei commented on packages/app-store/wipemycalother/lib/reschedule.ts:
```diff
@@ -2,11 +2,11 @@ import type { TFunction } from "i18next";
 import logger from "@calcom/lib/logger";
 import { getTranslation } from "@calcom/lib/server/i18n";
-import { deleteMeeting } from "@calcom/features/conferencing/lib/videoClient";
 import prisma from "@calcom/prisma";
```
```suggestion
import { prisma } from "@calcom/prisma";
```

volnei commented on packages/trpc/server/routers/viewer/eventTypes/util.ts:
```diff
@@ -1,10 +1,10 @@
-import type { EventTypeRepository } from "@calcom/features/eventtypes/repositories/eventTypeRepository";
 import { UserRepository } from "@calcom/features/users/repositories/UserRepository";
+import { markdownToSafeHTML } from "@calcom/lib/markdownToSafeHTML";
 import prisma from "@calcom/prisma";
```
```suggestion
import { prisma } from "@calcom/prisma";
```

devin-ai-integration[bot] committed:
fix: add deleted: null filter to bookingReference update operations

- Add deleted: null filter to API v1 PATCH endpoint to prevent updating soft-deleted booking references
- Add deleted: null filter to DailyVideo updateMeetingTokenIfExpired and setEnableRecordingUIAndUserIdForOrganizer
- Add comprehensive test coverage for PATCH endpoint soft-delete behavior
- Tests verify that soft-deleted booking references cannot be updated
- Tests verify that only active booking references can be updated successfully

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

keithwillcode commented on packages/app-store/dailyvideo/lib/VideoApiAdapter.ts:
```diff
@@ -185,6 +185,7 @@ export const updateMeetingTokenIfExpired = async ({
     await prisma.bookingReference.update({
       where: {
         id: bookingReferenceId,
+        deleted: null,
```
DevinAI, remove this. We are preserving functionality in this PR.

devin-ai-integration[bot] committed:
revert: remove deleted: null filters to preserve existing functionality

Per @keithwillcode's feedback, reverting the soft-delete filtering changes to preserve existing functionality in this PR. This PR should focus only on the Prisma upgrade itself.

- Reverted API v1 PATCH endpoint change
- Reverted DailyVideo adapter changes (updateMeetingTokenIfExpired and setEnableRecordingUIAndUserIdForOrganizer)
- Removed test file that was added for soft-delete behavior testing

Addresses comments:
- https://github.com/calcom/cal.com/pull/23816#discussion_r2448854197
- https://github.com/calcom/cal.com/pull/23816#discussion_r2448860594
- https://github.com/calcom/cal.com/pull/23816#discussion_r2448860833

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

devin-ai-integration[bot] committed:
test: restore and update booking reference tests to match existing functionality

Updated tests to verify existing behavior where PATCH endpoint can update
booking references regardless of their deleted status. This matches the
current implementation after reverting the deleted: null filters.

Changes:
- Restored test file that was previously deleted
- Updated PATCH tests to expect successful updates of soft-deleted references
- Renamed test suite to 'Existing functionality' to clarify intent
- Tests now verify that the PATCH endpoint preserves existing behavior

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

volnei commented on packages/features/routing-forms/lib/getRoutedUrl.ts:
```diff
@@ -16,17 +16,18 @@ import { substituteVariables } from "@calcom/app-store/routing-forms/lib/substit
 import { withReporting } from "@calcom/lib/sentryWrapper";
 import { PrismaRoutingFormRepository } from "@calcom/lib/server/repository/PrismaRoutingFormRepository";
-import { UserRepository } from "@calcom/features/users/repositories/UserRepository";
 import prisma from "@calcom/prisma";
```
```suggestion
import { prisma } from "@calcom/prisma";
```

devin-ai-integration[bot] committed:
test: rename booking-references test to integration-test

The test requires a database connection and should run in the integration
test job, not the unit test job. Renamed from .test.ts to .integration-test.ts
to match the repository's testing conventions.

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>
